modules/tile-converter/src/3d-tiles-converter/helpers/3d-tiles-content-converter.ts

```diff
@@ -117,16 +138,132 @@ export default class B3dmConverter {
+      featureAttributes,
+      attributeStorageInfo
+    );
+    const tableIndex = createExtStructuralMetadata(gltfBuilder, propertyAttributes);
 
-    return gltfBuffer;
+    const mesh = gltfBuilder.getMesh(meshIndex);
+    for (const primitive of mesh.primitives) {
+      if (tileContent.attributes._BATCHID?.value) {
+        createExtMeshFeatures(
+          gltfBuilder,
+          primitive,
+          tileContent.attributes._BATCHID.value,
+          tableIndex
+        );
+      }
+    }
+  }
+
+  private _createPropertyAttibutes(
+    featureAttributes: I3STileAttributes | null,
+    attributeStorageInfo?: AttributeStorageInfo[] | null | undefined
+  ): PropertyAttribute[] {
+    if (!featureAttributes || !attributeStorageInfo) {
+      return [];
+    }
+    const propertyAttributeArray: PropertyAttribute[] = [];
+    for (const attributeName in featureAttributes) {
+      const propertyAttribute = this._convertAttributeStorageInfoToPropertyAttribute(
+        attributeName,
+        attributeStorageInfo,
+        featureAttributes
+      );
+      if (propertyAttribute) {
+        propertyAttributeArray.push(propertyAttribute);
+      }
+    }
+    return propertyAttributeArray;
+  }
+
+  // eslint-disable-next-line complexity
+  private _convertAttributeStorageInfoToPropertyAttribute(
+    attributeName: string,
+    attributeStorageInfo: AttributeStorageInfo[],
+    featureAttributes: I3STileAttributes
+  ): PropertyAttribute | null {
+    const attributeValues = featureAttributes[attributeName];
+    const info = attributeStorageInfo.find((e) => e.name === attributeName);
+    if (!info) {
+      return null;
+    }
+    const attributeMetadata = info.attributeValues;
+    if (!attributeMetadata?.valueType) {
+      return null;
+    }
+    let elementType: string;
+    let componentType: string | undefined;
+    switch (attributeMetadata.valueType.toLowerCase()) {
+      case 'oid32':
+        elementType = 'SCALAR';
+        componentType = 'UINT32';
+        break;
+      case 'int32':
+        elementType = 'SCALAR';
+        componentType = 'INT32';
+        break;
+      case 'uint32':
+        elementType = 'SCALAR';
+        componentType = 'UINT32';
+        break;
+      case 'int16':
+        elementType = 'SCALAR';
+        componentType = 'INT16';
+        break;
+      case 'uint16':
+        elementType = 'SCALAR';
+        componentType = 'UINT16';
+        break;
+      case 'float64':
+        elementType = 'SCALAR';
+        componentType = 'FLOAT64';
+        break;
+      case 'string':
+        elementType = 'STRING';
+        break;
+      default:
+        elementType = '';
+        break;
+    }
+    const propertyAttribute: PropertyAttribute = {
+      name: attributeName,
+      elementType,
+      componentType,
+      values: []
+    };
+    if (isTypedArray(attributeValues)) {
+      propertyAttribute.values = Array.prototype.slice.call(attributeValues);
```

Comment:
I don't think this code is correct. `Function.prototype.call()` expects `thisArg` as the first parameter. Use `Array.from(attributeValues)` to convert a typed array to a JS array